Kitodo-MODSXML-SchemaConverter/src/main/java/org/kitodo/modsxmlschemaconverter/ModsXMLSchemaConverter.java

```diff
@@ -0,0 +1,138 @@
+import java.util.UnknownFormatConversionException;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Result;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.sax.SAXResult;
+import javax.xml.transform.sax.SAXSource;
+import javax.xml.transform.sax.SAXTransformerFactory;
+import javax.xml.transform.sax.TransformerHandler;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kitodo.api.schemaconverter.DataRecord;
+import org.kitodo.api.schemaconverter.FileFormat;
+import org.kitodo.api.schemaconverter.MetadataFormat;
+import org.kitodo.api.schemaconverter.SchemaConverterInterface;
+import org.kitodo.exceptions.ConfigException;
+import org.xml.sax.InputSource;
+
+public class ModsXMLSchemaConverter implements SchemaConverterInterface {
+
+    private static final Logger logger = LogManager.getLogger(ModsXMLSchemaConverter.class);
+
+    private static final String XSLT_FILEPATH = "/xslt/mods2kitodo.xsl";
+
+    private static MetadataFormat supportedSourceMetadataFormat = MetadataFormat.MODS;
+    private static MetadataFormat supportedTargetMetadataFormat = MetadataFormat.KITODO;
+    private static FileFormat supportedSourceFileFormat = FileFormat.XML;
+    private static FileFormat supportedTargetFileFormat = FileFormat.XML;
+
+    /**
+     * Converts a given DataRecord to the given MetadataFormat 'targetMetadataFormat' and FileFormat 'targetFileFormat'.
+     *
+     * @param record DataRecord to be converted
+     * @param targetMetadataFormat MetadataFormat to which the given DataRecord is converted
+     * @param targetFileFormat FileFormat to which the given DataRecord is converted
+     * @return The result of the conversion as a DataRecord.
+     */
+    @Override
+    public DataRecord convert(DataRecord record, MetadataFormat targetMetadataFormat, FileFormat targetFileFormat) {
+        if (!(supportsSourceMetadataFormat(record.getMetadataFormat())
+                && supportsSourceFileFormat(record.getFileFormat())
+                && supportsTargetMetadataFormat(targetMetadataFormat)
+                && supportsTargetFileFormat(targetFileFormat))) {
+            throw new UnknownFormatConversionException("Unable to convert from " + record.getMetadataFormat() + "/"
+                    + record.getFileFormat() + " to " + targetMetadataFormat + "/" + targetFileFormat + "!");
+        }
+
+        if (record.getOriginalData() instanceof InputStream) {
+            try (InputStream inputStream = (InputStream)record.getOriginalData();
+                 InputStream fileStream = getClass().getResourceAsStream(XSLT_FILEPATH)) {
```

Comment:
Did I interpret this correct, that this module can only handle one xsl for all possible transformation? If I'm correct then every catalogue which is using mods encoded data can or must relay on one transformation file? Why not add this file as a parameter to the method and let the usage of this module decide which transformation file is used.

This is similar to the MODS catalogue usage in 2.x but there you have the possibility to define a transformation file per catalogue and this is heavy used at SLUB.